scripts/01_directional_clustering.py

```diff
@@ -257,34 +260,30 @@
 # below there is the simplest snippet, but you can see more stuff
 # in the scripts/visualization folder
 
-# ClusterPlotter is a custom wrapper around a COMPAS MeshPlotter
-# the COMPAS MeshPlotter is built atop of pure Matplotlib (which is crazy)
-# what is different here is that I extended the plotter so that it can plot
-# vector fields directly as little lines via
-# ClusterPlotter.draw_vector_field_array()
-plotter = ClusterPlotter(mesh, figsize=(12, 9))
+# PlyPlotter is a custom wrapper around a Plotly graph object (Figure)
+# that handles formating and adjusts data structure.
+plotter = PlyPlotter() # plotly
 
 # draw only the boundary edges of the COMPAS Mesh
-plotter.draw_edges(keys=list(mesh.edges_on_boundary()))
+# plotter.draw_edges(keys=list(mesh.edges_on_boundary())) # matplotlib
+# TODO: add the equivalent of this function to ply_plotters
 
 if draw_faces:
     # color up the faces of the COMPAS mesh according to their cluster
-    # make a dictionary with all labels
-    labels_to_color = {}
-    for fkey in mesh.faces():
-        labels_to_color[fkey] = mesh.face_attribute(key=fkey, name="cluster")
-    # convert labels to rgb colors
-    face_colors = rgb_colors(labels_to_color, invert=False)
-    # draw faces
-    plotter.draw_faces(facecolor=face_colors)
+    plotter.draw_trimesh(mesh, paint_clusters, draw_mesh_edges)
 
 # draw vector fields on mesh as lines
 if draw_vector_fields:
-    # original vector field
-    va = vectors  # shorthand
-    plotter.draw_vector_field_array(va, (50, 50, 50), True, 0.07, 0.5)
+    # # original vector field
+    plotter.draw_vector_field_array(mesh, vectors, (50, 50, 50), True, 0.07, 0.5)
```

Comment:
Yes, `vectors` is a `VectorField()`. This comment made me look into `vector_field.py` which I now see now includes an option `.to_sequence` which probably makes some of the code inside the functions redundant. I'll adjust that.

The difference is in visualization. The cones make it easy to plot vector fields and require less processing of data, but they show a clear preferred direction or side. `draw_vector_field_array` outputs the same lines from previous plots, which only show lines without a preferred direction/side. I included both options because, depending on the discussion you might want to have (scientifically), the representation of vectors might matter in a given context. I think it would be best to change the name `draw_vector_field_array` to `draw_vector_field_lines`

Reply:
Oh no! I found another error from the `VectorField()` type. I don't think I picked it up because I was working on a different commit before and made the updates on the new commit assuming the types would not have changed.

Reply:
I adjusted the changes to the expected type(VectorField()), but I don't think the `to_sequence` method would resolve what I needed for the plot, so I left the additional "structural" step unchanged. 
**Question**: how much data structure changes do we incorporate as functions inside other files (such as plotters) and how much should be incorporated as class methods? -> this is not a suggestion to change anything at this point, but a question to learn from if I need to do smth like this in the future.

What I'm thinking is: if I need to access the same structure from other files apart from `plotters`, then it is clearer, to me, that the data structure methods should be accessible in the class `VectorField()` (right?). What I'm unsure is: if we are only accessing these data structures from one file - such as plotters- over and over again... then do we leave these adaptations within `plotters` only..? 

Rephrasing in a more general sense: how is data structure managed in this type of project that we are learning in this course? ...through the "main class" that originated the data, or through other added functions?  @arpastrana @DbCrWk (Dev, lmk if you need more context)

Reply:
My take is that if it is only used in one place (e.g. only inside one function), then keep it in that place as an _operation_. If the same functionality is used multiple times, across multiple functions, regardless of being only in `plotters`, then I put that functionality into a function or a class method @isabelmdo.